### PR TITLE
More factory functions

### PR DESCRIPTION
**Pull request by rbuckton**
This adds a number of missing factory functions for anyone writing custom transformers, as well as performs some cleanup in factory.ts.

Note, there are a few breaking API name changes in this PR as part of the public factory functions. I feel this is acceptable as custom transforms are still a fairly experimental feature, though if necessary we could export aliases for the renamed functions for backwards compatibility.

There's also a small change to the transform unit tests so that they aren't dependent on the new-line format of the host platform running the tests.

This builds on #15500 and #15511.

### CODE REVIEW

File: src/compiler/factory.ts [hunk 1]
```diff
@@ -281,378 +327,342 @@ namespace ts {
             : node;
     }
 
-    // Types
-
-    export function createKeywordTypeNode(kind: KeywordTypeNode["kind"]) {
-        return <KeywordTypeNode>createSynthesizedNode(kind);
+    export function createMethod(decorators: Decorator[] | undefined, modifiers: Modifier[] | undefined, asteriskToken: AsteriskToken | undefined, name: string | PropertyName, questionToken: QuestionToken | undefined, typeParameters: TypeParameterDeclaration[] | undefined, parameters: ParameterDeclaration[], type: TypeNode | undefined, body: Block | undefined) {
```
Comment (aozgaa) [inline src/compiler/factory.ts]: why make the `Declaration` part of the name implicit?

File: src/compiler/factory.ts [hunk 2]
```diff
@@ -281,378 +327,342 @@ namespace ts {
...
+            || node.decorators !== decorators
+            || node.modifiers !== modifiers
+            ? updateNode(createIndexSignature(decorators, modifiers, parameters, type), node)
             : node;
     }
 
-    export function createParenthesizedType(type: TypeNode) {
-        const node = <ParenthesizedTypeNode>createSynthesizedNode(SyntaxKind.ParenthesizedType);
+    /* @internal */
+    export function createSignatureDeclaration(kind: SyntaxKind, typeParameters: TypeParameterDeclaration[] | undefined, parameters: ParameterDeclaration[], type: TypeNode | undefined) {
```
Comment (aozgaa) [inline src/compiler/factory.ts]: In analogue to the above, should this be `createSignature`?

File: src/compiler/factory.ts [hunk 3]
```diff
@@ -214,61 +214,107 @@ namespace ts {
             : node;
     }
 
-    // Type Elements
+    // Signature elements
 
-    export function createSignatureDeclaration(kind: SyntaxKind, typeParameters: TypeParameterDeclaration[] | undefined, parameters: ParameterDeclaration[], type: TypeNode | undefined) {
```
Comment (aozgaa) [inline src/compiler/factory.ts]: I can't really tell from the diff if the code was merely re-ordered or if there are more substantive changes. Looks similar on the LHS and RHS though.

File: src/compiler/factory.ts (hunk 1 above)
Comment (rbuckton) [inline src/compiler/factory.ts]: Consistency with most other factory functions. Most of the time we're transforming to JavaScript, and JavaScript doesn't have method signatures.

File: src/compiler/factory.ts (hunk 2 above)
Comment (rbuckton) [inline src/compiler/factory.ts]: Per above, signatures are purely TypeScript. We can probably change the name if only to make it more concise.

File: src/compiler/factory.ts (hunk 3 above)
Comment (rbuckton) [inline src/compiler/factory.ts]: Mostly reordered to align with the order in `SyntaxKind`. It makes finding things easier.

**Pull request merged**